feat: Adiciona CRUD completo para Rascunhos (#14)

AgathaKarenne32 opened the pull request:
📋 Descrição
Este PR aplica o feedback da Camila e realiza uma série de melhorias e correções na funcionalidade de Rascunhos e na documentação da API.

Refatoração do Rascunho: O campo titulo foi renomeado para mercado em toda a feature (Entity, DTOs, Service) para refletir melhor seu propósito. Além disso, os campos userId e username foram removidos do RascunhoResponse.

Correção da UI do Swagger: Foi corrigido um bug na documentação interativa dos endpoints de autenticação (/logout, /refresh-token e /test-autenticated.). A adição da anotação @Parameter(hidden = true) no AuthController eliminou a redundância de campos de autorização, permitindo que os testes via navegador funcionem de forma correta e intuitiva usando apenas o botão "Authorize".Já que apartir do campo /login a adição do token em todos os campos seria desnecessário.

🔨 Tipo de mudança
Selecione uma ou mais opções:

[ ] Chore/Configuração (setup do projeto, ajustes técnicos, ambiente)  

[x] Refatoração (melhorias internas sem mudar funcionalidades)  

[ ] Nova funcionalidade (feature)  

[x] Bugfix (correção de um bug)  

[ ] Documentação  

[ ] Outro (especifique abaixo)

📝 Observações

-   Correção da UI do Swagger (Issue #20): Ajuste no `AuthController` com a anotação `@Parameter(hidden = true)` para eliminar a redundância de campos de autorização e permitir que os testes no navegador funcionem corretamente usando apenas o botão "Authorize".

Closes #14
Closes #20

Os endpoints do CRUD de Rascunhos continuam funcionando como antes, mas agora aceitam e retornam o campo mercado em vez de titulo, e a resposta não contém mais as informações do usuário. Os testes no Swagger foram refeitos e confirmaram o comportamento esperado.

AgathaKarenne32 committed:
feat: Adiciona CRUD completo para Rascunhos (#14)

AgathaKarenne32 commented:
@ca-madureira

ca-madureira commented:
@AgathaKarenne32 vou testar nesta e na proxima semana.

ca-madureira reviewed (dismissed):
Na resposta nao retornar o usuario(userId e username), e no banco de dados trocar o titulo pelo banco do mercado(também alterar na request e response para mercado ao invés de titulo)

AgathaKarenne32 committed:
fix: Corrige envio de token no Swagger para endpoints de autenticação

AgathaKarenne32 committed:
fix(rascunho): Sincroniza Service com a nova estrutura do RascunhoResponse

O método paraRascunhoResponse em RascunhoServiceImpl ainda tentava passar os campos userId e username, que foram removidos do DTO RascunhoResponse.

Esta correção atualiza a chamada do construtor para corresponder à nova estrutura do DTO, resolvendo a falha de compilação.

SamuelLimaCap commented on src/main/java/com/prati/projetomercado/service/impl/RascunhoServiceImpl.java:
```diff
@@ -0,0 +1,83 @@
+    }
+
+    @Override
+    public void apagarRascunho(Long rascunhoId) {
```
Estou conseguindo deletar rascunho criado na CONTA1 utilizando uma CONTA2. Isso não pode acontecer. Cada usuário só pode deletar seu próprio rascunho

SamuelLimaCap commented on src/main/java/com/prati/projetomercado/service/impl/RascunhoServiceImpl.java:
```diff
@@ -0,0 +1,83 @@
+    }
+
+    @Override
+    public RascunhoResponse atualizarRascunho(Long rascunhoId, UpdateRascunhoRequest updateRascunhoRequest) {
```
Se eu criar um rascunho na CONTA1, a CONTA2 consegue editar o rascunho da conta 1. Isso não deve acontecer. Cada conta só pode gerenciar seus próprios rascunhos

SamuelLimaCap reviewed (dismissed):
Se der, aproveita e atualiza sua branch com base na develop. Eu acabei de fazer um hotfix em relação a conta google. Corrigir isso vai ajudar a agatha a testar mais rápido

SamuelLimaCap committed:
Verifica se o usuário tem esse rascunho antes de editar/deletar

SamuelLimaCap committed:
Merge branch 'develop' into feat/14/crud-rascunhos
